### PR TITLE
k6 v0.32

### PR DESCRIPTION
**Pull request by ppcano**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/k6io/k6/releases/tag/v0.32.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/k6/encoding.d.ts [hunk 1]
```diff
@@ -7,8 +7,24 @@
...
+ * Base64 decode a string.
+ * https://k6.io/docs/javascript-api/k6-encoding/b64decode-input-encoding
+ * @param input - Base64 encoded string or ArrayBuffer object.
+ * @param encoding - Base64 variant.
+ * @returns Decoded string.
+ * @example
+ * encoding.b64decode(str)
+ * encoding.b64decode(str, 'rawstd')
+ * const decBuffer = encoding.b64decode(str, 'rawurl')
+ * let decBin = new Uint8Array(decBuffer);
```
Comment (na--) [inline types/k6/encoding.d.ts:25]: The `format` parameter isn't described here or in the example

File: types/k6/encoding.d.ts [hunk 2]
```diff
@@ -41,8 +57,23 @@ declare namespace encoding {
...
+     * Base64 decode a string.
+     * https://k6.io/docs/javascript-api/k6-encoding/b64decode-input-encoding/
+     * @param input - Base64 encoded string or ArrayBuffer object.
+     * @param encoding - Base64 variant.
+     * @returns Decoded string.
+     * @example
+     * encoding.b64decode(str)
+     * encoding.b64decode(str, 'rawstd')
+     * const decBuffer = encoding.b64decode(str, 'rawurl')
+     * let decBin = new Uint8Array(decBuffer);
```
Comment (na--) [inline types/k6/encoding.d.ts:74]: Same here

**Review by na-- — COMMENTED** [review]
LGTM besides the minor nitpick, but @imiric should probably also take a look

File: types/k6/encoding.d.ts (hunk 2 above)
Comment (ppcano) [reply types/k6/encoding.d.ts]: should be fixed by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53101/commits/daf9b4e88aede66d57bc0721af49ec5c95616c89

File: types/k6/encoding.d.ts (hunk 1 above)
Comment (ppcano) [reply types/k6/encoding.d.ts]: should be fixed by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53101/commits/daf9b4e88aede66d57bc0721af49ec5c95616c89

File: types/k6/encoding.d.ts [hunk 3]
```diff
@@ -1,14 +1,30 @@
 /**
  * Base64 decode a string.
- * https://k6.io/docs/javascript-api/k6-encoding/b64decode-input-encoding
- * @param input - Base64 encoded string or ArrayBuffer object.
+ * https://k6.io/docs/javascript-api/k6-encoding/b64decode-input-encoding-format/
+ * @param input - The string to base64 decode.
  * @param encoding - Base64 variant.
  * @returns Decoded string.
```
Comment (imiric) [inline types/k6/encoding.d.ts:6]: Hhmm why is `b64decode()` repeated? We need separate versions with and without the `format` argument?

The `@returns` description here should probably match the one below, though.

File: types/k6/index.d.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for k6 0.31
+// Type definitions for k6 0.32
```
Comment (imiric) [inline types/k6/index.d.ts:1]: Nit: the actual version is `0.32.0`, we shouldn't leave out the patch portion even if it's `0`.

**Review by imiric — APPROVED** [review]
I don't use TypeScript so I'm not very familiar with these definitions, but the changes LGTM.

File: types/k6/encoding.d.ts (hunk 3 above)
Comment (ppcano) [reply types/k6/encoding.d.ts]: > We need separate versions with and without the format argument

I think we need this to return the correct type depending on the format parameter. See the tests https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53101/commits/d5387b709bc015bab366bccffc496839b36da955


> The @returns description here should probably match the one below, though.

done https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53101/commits/d2653b966428c44f5f0218b782dcd0b516f9fca2

**Review by imiric — APPROVED** [review]

**Comment by ppcano** [comment]
Ready to merge

**Pull request merged**